File: DelegationDesignPattern/src/main/java/org/example/java/Room.java
package org.example.java;

public class Room implements WindowActionListener,DoorActionListener{
    private final WindowActionListener window=new Window();
    private final DoorActionListener door=new Door();
    public void closeDoor(){
        door.closeDoor();
    }
    public void openDoor(){
        door.openDoor();
    }
    public void closeWindow(){
      window.closeWindow();
    }
    public void openWindow(){
        window.openWindow();
    }

}

File: DelegationDesignPattern/src/main/java/org/example/kotlin/Room.kt
package org.example.kotlin

import org.example.java.WindowActionListener


class Room: WindowActionListener, DoorActionListener {
    private val window= Window()
    private val door= Door()
    override fun openWindow() {
       window.openWindow()
    }

    override fun closeWindow() {
      window.closeWindow()
    }

    override fun openDoor() {
        door.openDoor()
    }

    override fun closeDoor() {
        door.closeDoor()
    }

}